### ISSUE TITLE
Weight initialization

### ISSUE THREAD

**Issue by maddataanalyst** [masterhudini/info-spillover#10]
When building models - you should use some weights initializers with random seed.

[Torch initializers](https://docs.pytorch.org/docs/stable/nn.init.html).
This ensures deterministic training.

In module: `src/modules/models/static_gnn.py`

in part:

```python

class StaticGNNClassifier(nn.Module):
    """MLP encoder + static-weight GNN + graph-level classifier."""

    def __init__(
        self,
        num_features: int,
        hidden_dim: int,
        gnn_dim: int,
        num_classes: int,
        adjacency: torch.Tensor,
    ):
        super().__init__()
        if adjacency.dim() != 2 or adjacency.size(0) != adjacency.size(1):
            raise ValueError("Adjacency must be square (N x N)")

        self.register_buffer("adjacency", adjacency)

        self.encoder = nn.Sequential(
            nn.Linear(num_features, hidden_dim), #INIITIALIZER HERE!!!
            nn.ReLU(),
        )
        self.gnn_linear = nn.Linear(hidden_dim, gnn_dim, bias=False) #INIITIALIZER HERE!!!
        self.self_linear = nn.Linear(hidden_dim, gnn_dim) #INIITIALIZER HERE!!!
        self.classifier = nn.Linear(gnn_dim, num_classes) #INIITIALIZER HERE!!!
```
